### PR TITLE
ZEPPELIN-866] Hotfix  -  does not working search box on navbar.

### PR DESCRIPTION
**Pull request by cloverhearts**
### What is this PR for?
When you type in the search box and start the search,
The result is not exposed.
Previously, it was working well.

### What type of PR is it?
Hot Fix

### Todos
* [x] - fixed bug on navbar.html and controller

### What is the Jira issue?
https://issues.apache.org/jira/browse/ZEPPELIN-866
### How should this be tested?
try to search action on navbar.

#### before 
test02

#### after
test01

### Questions:
* Does the licenses files need update? no
* Is there breaking changes for older versions? no
* Does this needs documentation? no

### CODE REVIEW

File: zeppelin-web/src/components/navbar/navbar.controller.js [hunk 1]
```diff
@@ -91,8 +91,9 @@ angular.module('zeppelinWebApp').controller('NavCtrl', function($scope, $rootSco
       });
   };
 
-  $scope.search = function() {
-    $location.url(/search/ + $scope.searchTerm);
+  $scope.search = function(searchTerm) {
+    console.log('search term ', searchTerm);
```
Comment (prabhjyotsingh) [inline zeppelin-web/src/components/navbar/navbar.controller.js]: Can you remove console.log from here.

Comment (cloverhearts) [inline zeppelin-web/src/components/navbar/navbar.controller.js]: @prabhjyotsingh oh, Thank you!

**Comment by prabhjyotsingh** [comment]
If we use `$scope.search = function(searchTerm)` then we can [this](https://github.com/apache/incubator-zeppelin/blob/master/zeppelin-web/src/components/navbar/navbar.controller.js#L49-56) function becomes redundant. Can you remove this as well.



    $rootScope.$on('$locationChangeSuccess', function () {
     var path = $location.path();
     // hacky solution to clear search bar
     // TODO(felizbear): figure out how to make ng-click work in navbar
     if (path === '/') {
       $scope.searchTerm = '';
     }
    });

**Comment by cloverhearts** [comment]
@prabhjyotsingh 
I think that code would clean up the search box.
It should not be deleted.

**Comment by prabhjyotsingh** [comment]
Or that is also broken for `$locationChangeSuccess`, its not cleaning up. Can you fix this as well ?

**Comment by cloverhearts** [comment]
@prabhjyotsingh Okay, i will look to this issue.

**Comment by cloverhearts** [comment]
@prabhjyotsingh  You're all right.
you're alright.
this is code is bug and dead code. If active to code, the search becomes impossible.
So I have deleted that code.

**Comment by cloverhearts** [comment]
@prabhjyotsingh 
fix done.
Please, Can you feedback for this pr for me?

**Comment by bzz** [comment]
Thanks for hotfix, but do we know why it stopped working in the first place?
Let me take a look.

\cc @felizbear for review as well

**Comment by astroshim** [comment]
I tesed and working well. but it seems need a "initial statement".

File: zeppelin-web/src/components/navbar/navbar.html [hunk 2]
```diff
@@ -65,7 +65,7 @@
         <!--TODO(bzz): move to Typeahead https://angular-ui.github.io/bootstrap  -->
           <form role="search"
             style="width: 300px; display: inline-block; margin: 0 10px"
-             ng-submit="search()">
+             ng-submit="search(searchTerm)">
```
Comment (felizbear) [inline zeppelin-web/src/components/navbar/navbar.html]: i definitely like passing the argument to `search` instead of taking it from the `$scope` 👍

**Comment by cloverhearts** [comment]
@astroshim Thank you for feed back.
are you mean 'initial statement' is 'clean up on search box'?

**Comment by AhyoungRyu** [comment]
@cloverhearts I agree with @astroshim. I think he mean when the result is nothing, Zeppelin needs to show some statement like "We couldn’t find any notebook matching 'something_you_typed' "

**Comment by astroshim** [comment]
Thanks for the detail explanation AhyoungRyu.

**Comment by cloverhearts** [comment]
Thank you for explaining, @astroshim @AhyoungRyu 
I agree too.
But, This is must exist only for bug fixes at this PR.
I will make it in the next PR.

**Comment by AhyoungRyu** [comment]
@cloverhearts Yeah you're right. LGTM 👍

**Comment by prabhjyotsingh** [comment]
I think this sounds good, have created a jira for the same https://issues.apache.org/jira/browse/ZEPPELIN-869. LGTM. 
I'll Merge this if no more discussion.

**Comment by bzz** [comment]
> Let me take a look.

@prabhjyotsingh could you please wait, as there was an issue, but not addressed yet?

> Thanks for hotfix, but do we know why it stopped working in the first place?
It would be really cool, if we could know

**Comment by prabhjyotsingh** [comment]
Surething.

**Comment by cloverhearts** [comment]
@bzz 
can not access to '$ scope.searchTerm variable at navbar.html' in navbar.controller
Code and nav are not currently operating normally.
I did not find the PR of the problem, which seems to be a problem at some point.
This can cause a lot of problems in the current navbar.
For example
https://github.com/apache/incubator-zeppelin/pull/912

I think we need to recover the ability to hotfix first.
And it should retouching the navbar.

**Comment by bzz** [comment]
@cloverhearts fix looks great to me. Pity that we can not identify the source of the bug though.

Let's merge as a hotfix!

**Comment by prabhjyotsingh** [comment]
Sure, I'll merge this.

**Pull request closed without merging**